objectivec/include/ort_session.h

```diff
@@ -209,6 +210,26 @@ NS_ASSUME_NONNULL_BEGIN
 - (BOOL)registerCustomOpsUsingFunction:(NSString*)registrationFuncName
                                  error:(NSError**)error;
 
+/**
+ * Registers custom ops for use with `ORTSession`s using this SessionOptions by calling the specified function
+ * pointed to by `registerCustomOpsFn`.
+ *
+ * Available since 1.16.
+ *
+ * The registration function must have the signature:
+ *    `OrtStatus* (*fn)(OrtSessionOptions* options, const OrtApiBase* api);`
+ *
+ * See https://onnxruntime.ai/docs/reference/operators/add-custom-op.html for more information on custom ops.
+ * See https://github.com/microsoft/onnxruntime/blob/342a5bf2b756d1a1fc6fdc582cfeac15182632fe/onnxruntime/test/testdata/custom_op_library/custom_op_library.cc#L115
+ * for an example of a custom op library registration function.
+ *
+ * @param registerCustomOpsFn A pointer to the registration function to call.
+ * @param error Optional error information set if an error occurs.
+ * @return Whether the registration function was successfully called.
+ */
+- (BOOL)registerCustomOpsUsingFunctionPointer:(ORTCAPIRegisterCustomOpsFnPtr)registerCustomOpsFn
+                                        error:(NSError**)error;
```

Comment:
Should we explicitly say the caller is responsible for validating the pointer as there's no way we can do that? If someone is assessing security risks it would make it clear this function needs careful consideration.

Could we mention the function signature is defined in the ORT C API
https://github.com/microsoft/onnxruntime/blob/67f4cd54fab321d83e4a75a40efeee95a6a17079/include/onnxruntime/core/session/onnxruntime_c_api.h#L697 so the link between the two is clear?

Reply:
good idea, updated the docs